### ISSUE TITLE
Callback trigger using `options.variables`

### ISSUE THREAD

**Issue by freitasskeeled** [jhildenbiddle/css-vars-ponyfill#59]
Hi,

Is there any way of using `options.variables` with the remaining options set to the default values and trigger ponyfill callbacks like `onComplete`?

The problem is that knowing when the update finishes is useful and using `options.variables` removes that option since there is no way to apply those changes only in legacy.

Note: obviously, I could set `options.onlyLegacy` to false but I don't want to loose performance in modern browsers.

Thank you.

**Comment by jhildenbiddle** [comment]
Hi @freitasskeeled.

TL;DR: No, but the limitation is easy to work around (see **Workaround** below).

**Explanation**

The reason why the `onComplete` callback is not invoked when `onlyLegacy` is `true` is to avoid the potential confusion caused by unintentionally executing code intended for legacy browsers in modern as well.

For example, most devs would assume the following call would only affect legacy browsers:

```js
cssVars({
  onlyLegacy: true, // default
  onComplete() {
    // Do more legacy browser stuff...
  }
});
```

Given the ponyfill's current behavior, they would be correct. If the `onComplete` callback above were to be invoked just because `options.variables` is passed:

```js
cssVars({
  onlyLegacy: true, // default
  variables: {
    "--color1": "red",
    "--color2": "green",
    "--color3": "blue"
  },
  onComplete() {
    // Do more legacy browser stuff...
  }
});
```

... things start to get confusing very quickly because 1) the `// Do more legacy browser stuff...` code is now executing in modern browsers, and 2) sometimes the callback fires in modern browsers (when `options.variables` are passed) and other times it doesn't. 

Devs could prevent this by testing for native CSS custom property support before executing their legacy code:

```js
onComplete() {
  const hasNativeCssVarsSupport = window.CSS && window.CSS.supports && window.CSS.supports('(--a: 0)');

  if (!hasNativeCssVarsSupport) {
    // Do more legacy browser stuff...
  }
}
```

... but it isn't obvious that they would need to do this, and it's understandably confusing why this type of check needs to done when `onlyLegacy` is set to `true`.

**Workaround**

When `onlyLegacy` is `true`, custom properties passed via `options.variables` are updated synchronously in modern browsers via the [style.setStyle()](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration/setProperty) method so you don't need to wait for a callback. If you have post-ponyfill code that needs to execute in both browsers, write that code in a separate function that can be called 1) from the ponyfill callback for legacy browsers and 2) after the callback in a modern-browser-only if-statement:

```js
// Test for native CSS custom property support
const hasNativeCssVarsSupport = window.CSS && window.CSS.supports && window.CSS.supports('(--a: 0)');

// Create a function that performs post-ponyfill tasks
function doStuff() {
  // Do stuff in modern & legacy browsers...
}

// Call the ponyfill to update custom property values
cssVars({
  onlyLegacy: true, // default
  variables: {
    "--color1": "red",
    "--color2": "green",
    "--color3": "blue"
  },
  // Do stuff in legacy browsers
  onComplete(): doStuff
});

// Do stuff in modern browsers
if (hasNativeCssVarsSupport) {
  doStuff();
}
```

Hope this helps.
